pkg/controllers/elasticsearch/cluster_control.go

```diff
@@ -100,10 +100,50 @@ func NewController(
 	}
 }
 
+// checkPausedConditions checks if the given cluster is paused or not and adds an appropriate condition.
+func (e *defaultElasticsearchClusterControl) checkPausedConditions(c *v1alpha1.ElasticsearchCluster) error {
+	cond := c.Status.GetStatusCondition(v1alpha1.ClusterConditionProgressing)
+	pausedCondExists := cond != nil && cond.Reason == v1alpha1.PausedClusterReason
+
+	needsUpdate := false
+	if c.Spec.Paused && !pausedCondExists {
+		c.Status.UpdateStatusCondition(
+			v1alpha1.ClusterConditionProgressing,
+			v1alpha1.ConditionUnknown,
+			v1alpha1.PausedClusterReason,
+			"Cluster is paused",
+		)
+		needsUpdate = true
+	} else if !c.Spec.Paused && pausedCondExists {
+		c.Status.UpdateStatusCondition(
+			v1alpha1.ClusterConditionProgressing,
+			v1alpha1.ConditionUnknown,
+			v1alpha1.ResumedClusterReason,
+			"Cluster is resumed",
+		)
+		needsUpdate = true
+	}
+
+	if !needsUpdate {
+		return nil
+	}
+
+	var err error
+	c, err = e.navigatorClient.NavigatorV1alpha1().ElasticsearchClusters(c.Namespace).UpdateStatus(c)
+	return err
+}
+
```

Comment:
Maybe if this method instead mutated the Cluster Status in place and if there was an interface for cluster condition operations, it could be made into a generic function shared between cassandra and elasticsearch.
And the status is updated lower by `Controller.sync` lower in the stack, so we could avoid a separate API call.

Happy if you prefer to leave this for a  followup branch.